docs/source/spatialstats.rst

```diff
@@ -0,0 +1,143 @@
+
+Quantifying the precision of a single DEM, or of a difference of DEMs
+*********************************************************************
+
+To statistically infer the precision of a DEM, the DEM has to be compared against independent elevation observations.
+
+If the other elevation data is known to be of higher-precision, one can assume that the analysis of differences will represent the precision of the rougher DEM.
+Otherwise, significant measurement errors can originate from both sets of elevation observations, and the analysis of differences will represent the mixed precision of the two.
+
+TODO: complete with Hugonnet et al. (in prep)
+
+Stable terrain: proxy for infering DEM precision
+************************************************
+
+When comparing elevation datasets, stable terrain is usually used a proxy
+
+Metrics for DEM precision
+*************************
+
+The precision of DEMs has generally been reported as a single value indicating the random error at the scale of a single pixel, for example :math:`\pm 2` meters.
+
+However, the significant variability of elevation measurement errors
+
+Pixel-wise elevation measurement error
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+TODO
+
+
+Spatially-integrated elevation measurement error
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The standard error (SE) of a statistic is the standard deviation of the distribution of this statistic.
+For spatially distributed samples, the standard error of the mean (SEM) is of great interest as it allows quantification of the error of a mean (or sum) of samples in space.
+
+The standard error  :math:`\sigma_{\overline{dh}}` of the mean :math:`\overline{dh}` of elevation changes samples :math:`dh` is typically derived as:
+
+.. math::
+
+        \sigma_{\overline{dh}} = \frac{\sigma_{dh}}{\sqrt{N}},
+
+where :math:`\sigma_{dh}` is the dispersion of the samples, and :math:`N` is the number of **independent** observations.
+
+However, several issues arise to estimate the standard error of a mean of elevation observations samples:
+
+1. The dispersion :math:`\sigma_{dh}` cannot be estimated directly on changing terrain that we are usually interested in measuring (e.g., glacier, snow, forest).
+2. The dispersion :math:`\sigma_{dh}` typically shows important non-stationarities (e.g., an error 10 times as large on steep slopes than flat slopes).
+3. The number of samples :math:`N` is generally not equal to the number of sampled DEM pixels, as those are not independent in space and the Ground Sampling Distance of the DEM does not necessarily correspond to its effective resolution.
+
+Note that the SE represents completely stochastic (random) errors, and is therefore not accounting for possible remaining systematic errors have been accounted for, e.g. using one or multiple :ref:`coregistration` approaches.
+
+
+Workflow for DEM precision estimation
```

Comment:
This could also refer to an example.

I thought it was great to have these `.. literalinclude: \n :lines:: ...` directives at first.
A large problem that I've found is that they need maintenance whenever the code changes; something which is easy to miss.
I just fixed most of these blunders in the latest PRs, but there's one (tiny) blunder still in the Coregistration chapter:

https://xdem.readthedocs.io/en/latest/coregistration.html#introduction

In the bottom of the **Example data** code block, there's a "##################################" line which should not be visible.

I know that it was I that suggested this way of including code, so I'm sorry that I now recommend slightly against it... With the new examples structure, however, I think this issue can quite neatly be circumvented!

Reply:
+1, for all these examples, it would be great to have a separate code that shows in the example gallery!